activesupport/lib/active_support/core_ext/time/calculations.rb

```diff
@@ -159,6 +159,8 @@ def change(options)
       ::Time.new(new_year, new_month, new_day, new_hour, new_min, new_sec, new_offset)
     elsif utc?
       ::Time.utc(new_year, new_month, new_day, new_hour, new_min, new_sec)
+    elsif zone && zone.is_a?(ActiveSupport::TimeZone)
+      ::Time.new(new_year, new_month, new_day, new_hour, new_min, new_sec, zone)
```

Comment:
the original implementation merged this branch and the next one, but `Time.new` doesn't like zones formatted as some kinds of strings, eg. it was crashing on `::Time.new(new_year, new_month, new_day, new_hour, new_min, new_sec, "CST")`

Reply:
`time.zone` isn't necessarily an instance of `ActiveSupport::TimeWithZone` - Ruby allows any object there as long as it [quacks in the correct manner][1]. Perhaps a `zone.respond_to?(:utc_to_local)` would be better?

[1]: https://ruby-doc.org/core-3.0.1/Time.html#class-Time-label-Timezone+argument

Reply:
That's the docs I was trying to find! I'll tidy it up, thanks.